google-ads/src/test/java/com/google/ads/googleads/lib/GoogleAdsClientTest.java

```diff
@@ -450,19 +419,43 @@ public void exceptionTransformedToGoogleAdsException() {
     }
   }
 
-  /**
-   * Ensure that can set endpoint on default transport channel.
-   */
+  /** Verifies that the exception transformation behaviour is working for a test example. */
+  @Test
+  public void streamingCallable_exceptionTransformedToGoogleAdsException() {
+    GoogleAdsClient client =
+        GoogleAdsClient.newBuilder()
+            .setCredentials(fakeCredentials)
+            .setDeveloperToken(DEVELOPER_TOKEN)
+            .setTransportChannelProvider(localChannelProvider)
+            .setEnableGeneratedCatalog(enabledGeneratedCatalog)
+            .build();
+    Metadata.Key trailerKey =
+        ApiCatalog.getDefault().getLatestVersion().getExceptionFactory().getTrailerKey();
+    Metadata trailers = new Metadata();
+    GoogleAdsFailure.Builder failure = GoogleAdsFailure.newBuilder();
+    failure.addErrors(GoogleAdsError.newBuilder().setMessage("Test error message"));
+    trailers.put(trailerKey, failure.build().toByteArray());
+    StatusException rootCause = new StatusException(Status.UNKNOWN, trailers);
+    mockService.addException(new ApiException(rootCause, GrpcStatusCode.of(Code.UNKNOWN), false));
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        client.getLatestVersion().createGoogleAdsServiceClient()) {
+      googleAdsServiceClient
+          .searchStreamCallable()
+          .call(SearchGoogleAdsStreamRequest.getDefaultInstance());
+    } catch (GoogleAdsException ex) {
+      // Expected
```

Comment:
since this is expected, then please add a `fail()` call at the end of the `try` block

Reply:
Good call, done. Also needed to add a loop to attempt to process the results.